tensorflow_lite_support/cc/task/processor/image_postprocessor.cc

```diff
@@ -0,0 +1,129 @@
+using ::tflite::support::TfLiteSupportStatus;
+
+}  // namespace
+
+/* static */
+tflite::support::StatusOr<std::unique_ptr<ImagePostprocessor>>
+ImagePostprocessor::Create(core::TfLiteEngine* engine, const int output_index,
+                           const int input_index) {
+  ASSIGN_OR_RETURN(auto processor,
+                   Processor::Create<ImagePostprocessor>(
+                       /* num_expected_tensors = */ 1, engine, {output_index},
+                       /* requires_metadata = */ false));
+
+  RETURN_IF_ERROR(processor->Init(input_index));
+  return processor;
+}
+
+absl::Status ImagePostprocessor::Init(const int input_index) {
+  if (input_index == -1) {
+    return tflite::support::CreateStatusWithPayload(
+        absl::StatusCode::kInvalidArgument,
+        absl::StrFormat("Input image tensor not set. Input index found: %d",
+                        input_index),
+        tflite::support::TfLiteSupportStatus::kInputTensorNotFoundError);
+  }
+  ASSIGN_OR_RETURN(
+      auto output_specs,
+      vision::BuildImageTensorSpecs(*engine_->interpreter(),
+                                    *engine_->metadata_extractor(), false));
+  options_ = std::make_unique<vision::NormalizationOptions>(
+      output_specs.normalization_options.value());
+  return absl::OkStatus();
+}
+
+absl::StatusOr<vision::FrameBuffer> ImagePostprocessor::Postprocess() {
+  has_uint8_output_ = GetTensor()->type == kTfLiteUInt8;
+  const int kRgbPixelBytes = 3;
+
+  vision::FrameBuffer::Dimension to_buffer_dimension = {
+      GetTensor()->dims->data[2], GetTensor()->dims->data[1]};
+  size_t output_byte_size =
+      GetBufferByteSize(to_buffer_dimension, vision::FrameBuffer::Format::kRGB);
+  std::vector<uint8> postprocessed_data(output_byte_size / sizeof(uint8), 0);
+
+  if (has_uint8_output_) {  // No denormalization required.
+    if (GetTensor()->bytes != output_byte_size) {
+      return tflite::support::CreateStatusWithPayload(
+          absl::StatusCode::kInternal,
+          "Size mismatch or unsupported padding bytes between pixel data "
+          "and output tensor.");
+    }
+    const uint8* output_data =
+        core::AssertAndReturnTypedTensor<uint8>(GetTensor()).value();
+    postprocessed_data.insert(postprocessed_data.begin(), &output_data[0],
+                              &output_data[output_byte_size / sizeof(uint8)]);
+  } else {  // Denormalize to [0, 255] range.
+    if (GetTensor()->bytes / sizeof(float) !=
+        output_byte_size / sizeof(uint8)) {
+      return tflite::support::CreateStatusWithPayload(
+          absl::StatusCode::kInternal,
+          "Size mismatch or unsupported padding bytes between pixel data "
+          "and output tensor.");
+    }
+
+    uint8* denormalized_output_data = postprocessed_data.data();
+    const float* output_data =
+        core::AssertAndReturnTypedTensor<float>(GetTensor()).value();
+    const auto norm_options = GetNormalizationOptions();
+
+    if (norm_options.num_values == 1) {
```

Comment:
If you don't have a testing float model yet, this is a good one: 
https://github.com/margaretmz/Selfie2Anime-with-TFLite/blob/master/android/app/src/main/ml/selfie2anime.tflite
See the introduction of the model [here](https://github.com/margaretmz/Selfie2Anime-with-TFLite).

Reply:
The ESRGAN model works for both uint8 and float.